docs/connector-app-api.md

```diff
@@ -0,0 +1,165 @@
+# API exposed by the Smart Card Connector App
+
+The Smart Card Connector App exposes an API that allows other Chrome OS
+Extensions/Apps to make PC/SC calls (see the generic specification of this API
+at
+[https://pcsclite.apdu.fr/api/group__API.html](https://pcsclite.apdu.fr/api/group__API.html)).
+
+
+## Permissions for the PC/SC API consumers
+
+Due to various privacy and security concerns, the following decisions
+were made:
+
+1. When an Extension/App tries to talk to the Connector App, a **user prompt
+   dialog** is generally shown, asking whether to allow or to block this
+   Extension/App.
+
+   If the user decides to allow the Extension/App, the decision is remembered in
+   the Connector App's local storage, and the requests made by the Extension/App
+   start being actually executed by the Connector App. Otherwise, all the
+   requests made by the middleware App are refused.
+
+   Note: this system has nothing to do with the Chrome App permissions model
+   (see
+   [https://developer.chrome.com/apps/declare_permissions](https://developer.chrome.com/apps/declare_permissions)),
+   as Chrome allows to use the messaging API without any additional permissions.
+   This intent is to close the hole left open by cross-app messaging between
+   Apps with different permissions.
+
+2. The Connector App is **bundled with a whitelist of known Extension/App
+   identifiers** (and a mapping to their display names). For all Extensions/Apps
+   not from this list, the user prompt will contain a big scary warning message.
```

Comment:
Question: Is the big scary warning message the same thing as the user prompt dialog mentioned above? If so, points 1 and 2 can probably be merged.